Copyright (C) 2025 Eduardo Santos <eduardo.experimental@gmail.com>

This file is part of edcsnt/comp2.

edcsnt/comp2 is free software: you can redistribute it and/or modify it
under the terms of the GNU General Public License as published by the
Free Software Foundation, either version 3 of the License, or (at your
option) any later version.

edcsnt/comp2 is distributed in the hope that it will be useful, but
WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
Public License for more details.

You should have received a copy of the GNU General Public License along
with edcsnt/comp2. If not, see <https://www.gnu.org/licenses/>.
diff --git a/comp2.c b/comp2.c
index da33d01..e4ac0f2 100644
--- a/comp2.c
+++ b/comp2.c
@@ -34,9 +34,9 @@ main(void)
 	}
 	sigprocmask(SIG_UNBLOCK, &set, NULL);
 	while (1) {
-		fprintf(stderr,
-		        "wasting one file descriptor every 5s... :(\n");
-		open("./file", O_RDONLY | O_CREAT, S_IRUSR | S_IWUSR);
+		fprintf(stderr, "no more resource leaks! :)\n");
+		int fd = open("./file", O_RDONLY | O_CREAT, S_IRUSR | S_IWUSR);
+		close(fd);
 		sleep(5);
 	}
 }
